server-common/src/main/scala/com/aliyun/emr/rss/server/common/Service.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.server.common
+
+import io.netty.channel.ChannelFuture
+
+import com.aliyun.emr.rss.common.RssConf
+import com.aliyun.emr.rss.common.internal.Logging
+import com.aliyun.emr.rss.common.metrics.MetricsSystem
+import com.aliyun.emr.rss.server.common.http.{HttpRequestHandler, HttpServer, HttpServerInitializer, HttpServiceInterface}
+
+abstract class Service(conf: RssConf) extends Logging with HttpServiceInterface {
```

Comment:
Shall we merge Service and HttpServiceInterface? Currently, only Service extends HttpServiceInterface. Maybe we can consider splitting them when a different class needs to extend HttpServiceInterface in the future.

Reply:
> Shall we merge Service and HttpServiceInterface? Currently, only Service extends HttpServiceInterface. Maybe we can consider splitting them when a different class needs to extend HttpServiceInterface in the future.

Since `Service` should be a base class, how aboit current? `Service -> HttpService -> Master/Worker`